python/src/aac/plugins/_common.py

```diff
@@ -22,28 +34,82 @@ def get_plugin_definitions_from_yaml(package, filename) -> list[Definition]:
         return yaml_definitions
 
 
-def register_plugin_command(plugin_name: str, command_name: Optional[str] = None) -> Callable:
+def register_plugin_command(plugin_name: str, command_name: str) -> Callable:
     """
     Register a plugin command with the associated plugin.
 
     Args:
         plugin_name (str): The name of the plugin on which the command will be registered.
-        command_name (str): The name of the command to be registered. (default: plugin_name)
+        command_name (str): The name of the command to be registered.
     """
     global REGISTERED_PLUGIN_COMMANDS
 
-    command_name = command_name or plugin_name
-
     def wrapper(function: Callable):
+        if plugin_name in REGISTERED_PLUGIN_COMMANDS:
+            if command_name in REGISTERED_PLUGIN_COMMANDS[plugin_name]:
+                log.warn(f"Overwriting implementation of command {command_name} in plugin {plugin_name}")
+
+            REGISTERED_PLUGIN_COMMANDS[plugin_name].update({command_name: function})
+        else:
+            REGISTERED_PLUGIN_COMMANDS[plugin_name] = {command_name: function}
+
         @wraps(function)
         def wrapped(*args, **kwargs):
-            if plugin_name in REGISTERED_PLUGIN_COMMANDS:
-                REGISTERED_PLUGIN_COMMANDS[plugin_name].append(command_name)
-            else:
-                REGISTERED_PLUGIN_COMMANDS[plugin_name] = [command_name]
-
             return function(*args, **kwargs)
 
         return wrapped
 
     return wrapper
+
+
+def get_plugin_commands_from_definitions(plugin_name: str, plugin_definitions: List[Definition]) -> List[AacCommand]:
+    """
+    Return the AacCommands for the specified plugin.
+
+    Args:
+        plugin_name (str): The plugin name.
+        plugin_definitions (List[Definition]): A collection of definitions contributed by the plugin.
+    """
+
+    from aac.plugins.first_party.gen_plugin import DEFINITION_FIELD_NUMBER_OF_ARGUMENTS, DEFINITION_FIELD_DEFAULT
```

Comment:
I do realize this causes a dependency on the gen-plugin plugin but we need to come up with a much better approach to handling constants. We've discussed this previously but haven't come to a specific solution.

Reply:
We could probably capture this in _common or even a plugins/constants.py in the future